rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/scan/provider/UnzipArchivesToOutputRuleProvider.java

```diff
@@ -24,12 +38,48 @@
     @Override
     public Configuration getConfiguration(GraphContext context)
     {
-        return ConfigurationBuilder.begin().addRule()
-        .when(Query.fromType(ArchiveModel.class).excludingType(IgnoredArchiveModel.class))
-        .perform(
-            UnzipArchiveToOutputFolder.unzip(),
-            IterationProgress.monitoring("Unzipped archive", 1),
-            Commit.every(1)
-        );
+        return ConfigurationBuilder.begin()
+            .addRule()
+            .when(Query.fromType(ArchiveModel.class).excludingType(IgnoredArchiveModel.class))
+            .perform(
+                UnzipArchiveToOutputFolder.unzip(),
+                IterationProgress.monitoring("Unzipped archive", 1),
+                Commit.every(1)
+            )
+            .addRule()
+            .when(Query.fromType(ArchiveModel.class).excludingType(DuplicateArchiveModel.class))
+            .perform(new DuplicateArchiveOperation());
+    }
+
+    private class DuplicateArchiveOperation extends AbstractIterationOperation<ArchiveModel>
+    {
+
+        @Override
+        public void perform(GraphRewrite event, EvaluationContext context, ArchiveModel originalArchive)
+        {
+            // Skip if there were no duplicates
+            if (!originalArchive.getDuplicateArchives().iterator().hasNext()) {
+                return;
+            }
+
+            ArchiveModel originalParentArchive = originalArchive.getParentArchive();
+            FileModel originalArchiveParentFile = originalArchive.getParentFile();
+            originalArchive.setParentFile(null);
+            originalArchive.setParentArchive(null);
+
+            // Create the duplicate archive
+            GraphService<DuplicateArchiveModel> duplicateArchiveService = event.getGraphContext().service(DuplicateArchiveModel.class);
+            DuplicateArchiveModel duplicateArchive = duplicateArchiveService.create();
+            duplicateArchive.setOriginalArchive(originalArchive);
+            duplicateArchive.setSHA1Hash(originalArchive.getSHA1Hash());
+            duplicateArchive.setFilePath(originalArchive.getFilePath());
+            duplicateArchive.setArchiveName(originalArchive.getArchiveName());
+            duplicateArchive.setFileName(originalArchive.getFileName());
+            duplicateArchive.setParentArchive(originalParentArchive);
+            duplicateArchive.setParentFile(originalArchiveParentFile);
+
+            // Move the original archivemodel to the <shared-libs> application
+            duplicateArchive.setParentArchive(originalParentArchive);
```

Comment:
this looks like extra/leftover, you just set this four lines above, and the comment above it doesn't match